### PR TITLE
MGMT-12333: Add discovery kernel arguments support

### PR DESCRIPTION
**Pull request by ori-amizur**
Add capability to add discovery kernel arguments.  In case these arguments are added, they will be part of the infra-env creation payload.

/cc @eliorerz 
/cc @filanov

### CODE REVIEW

File: src/assisted_test_infra/test_infra/helper_classes/config/base_infra_env_config.py [hunk 1]
```diff
@@ -18,3 +18,4 @@ class BaseInfraEnvConfig(BaseEntityConfig, ABC):
     ignition_config_override: str = None
     verify_download_iso_ssl: bool = None
     is_static_ip: bool = None
+    discovery_kernel_arguments: List[dict] = None
```
Comment (osherdp) [inline src/assisted_test_infra/test_infra/helper_classes/config/base_infra_env_config.py:21]: nit, I guess it's rather:
```suggestion
    discovery_kernel_arguments: Optional[List[dict]] = None
```

**Comment by osherdp** [comment]
@ori-amizur how did you test this?

File: src/assisted_test_infra/test_infra/helper_classes/config/base_infra_env_config.py (hunk 1 above)
Comment (ori-amizur) [reply src/assisted_test_infra/test_infra/helper_classes/config/base_infra_env_config.py:21]: OK

**Comment by ori-amizur** [comment]
> how did you test this?

Created a cluster and saw the arguments in the hosts.  They can be seen by looking at the file /proc/cmdline

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Pull request merged**